libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -650,7 +650,7 @@ class MapboxNavigation @VisibleForTesting internal constructor(
     fun stopTripSession() {
         runIfNotDestroyed {
             latestLegIndex = tripSession.getRouteProgress()?.currentLegProgress?.legIndex
-            tripSession.stop()
+            tripSession.stop(true)
```

Comment:
As far as I understood from your discussion with @averkhaturau in the NN PR, it would only be true if we haven't arrived yet, right? If so, there's no such check here.

Reply:
The idea of the `cancelled : bool` parameter here is that routing session cancel is not a complete opposite to routing complete. E.g. User may finish the route but navigator will not detect it because the user parks on another side of the building / fence, etc. But actual session is successfully complete and the User shows it by pressing thumb up. From other side, session may be detected is finished by mistake, e.g. wrong map-matching, target point where is impossible to stop, etc., and the user will actually cancel the session.

Reply:
https://mapbox.atlassian.net/browse/NN-667